diff --git a/third_party/mediapipe/src/mediapipe/framework/tool/options_map.h b/third_party/mediapipe/src/mediapipe/framework/tool/options_map.h
index c35d6b26bf589..a856110bd8e05 100644
--- a/third_party/mediapipe/src/mediapipe/framework/tool/options_map.h
+++ b/third_party/mediapipe/src/mediapipe/framework/tool/options_map.h
@@ -73,7 +73,7 @@ void GetNodeOptions(const CalculatorGraphConfig::Node& node_config, T* result) {
 #else
   for (const mediapipe::protobuf::Any& options : node_config.node_options()) {
     if (options.Is<T>()) {
-      options.UnpackTo(result);
+      (void)options.UnpackTo(result);
     }
   }
 #endif
