diff --git a/src/google/protobuf/generated_message_tctable_impl.h b/src/google/protobuf/generated_message_tctable_impl.h
index d1e848a14c2f3..bb16390ed1694 100644
--- a/src/google/protobuf/generated_message_tctable_impl.h
+++ b/src/google/protobuf/generated_message_tctable_impl.h
@@ -892,14 +892,14 @@ class PROTOBUF_EXPORT TcParser final {
     const uint32_t has_bits_offset = table->has_bits_offset;
     if constexpr (internal::PerformDebugChecks()) {
       // We always have some offset to write to.
-      ABSL_DCHECK_NE(has_bits_offset, 0);
+      ABSL_DCHECK_NE(has_bits_offset, 0u);
       // and if we actually have has bits to push, we should be pushing to a
       // real HasBits.
       // `has_bits_offset` points to `_cached_size_` when we have
       // no has bits, just to point somewhere and avoid a branch here.
       // In that case we would be doing `|= 0` so the target just need to be
       // some valid space in the message.
-      if (static_cast<uint32_t>(hasbits) != 0) {
+      if (static_cast<uint32_t>(hasbits) != 0u) {
         ABSL_DCHECK_NE(has_bits_offset, table->class_data->cached_size_offset);
       }
     }
