diff --git a/third_party/protobuf/src/google/protobuf/port_def.inc b/third_party/protobuf/src/google/protobuf/port_def.inc
index b68efc85ac7ce..30e56d3dfb5c0 100644
--- a/third_party/protobuf/src/google/protobuf/port_def.inc
+++ b/third_party/protobuf/src/google/protobuf/port_def.inc
@@ -148,14 +148,17 @@ static_assert(PROTOBUF_ABSL_MIN(20230125, 3),
 // Owner: ckennelly@, mkruskal@
 #define PROTOBUF_FUTURE_ADD_NODISCARD [[nodiscard]]
 
+#define PROTOBUF_FUTURE_ADD_EARLY_NODISCARD [[nodiscard]]
+#define PROTOBUF_FUTURE_ADD_EARLY_WARN_UNUSED ABSL_ATTRIBUTE_WARN_UNUSED
+
 #else
 
 #define PROTOBUF_FUTURE_ADD_NODISCARD
+#define PROTOBUF_FUTURE_ADD_EARLY_NODISCARD
+#define PROTOBUF_FUTURE_ADD_EARLY_WARN_UNUSED
 
 #endif
 
-#define PROTOBUF_FUTURE_ADD_EARLY_NODISCARD [[nodiscard]]
-#define PROTOBUF_FUTURE_ADD_EARLY_WARN_UNUSED ABSL_ATTRIBUTE_WARN_UNUSED
 
 #ifdef PROTOBUF_ALWAYS_INLINE
 #error PROTOBUF_ALWAYS_INLINE was previously defined
